### PR TITLE
[ECOINT-305] Add live bandwidth metric 

### PR DESCRIPTION
**Pull request by BartoszBlizniak**
This pull request adds support for a new real-time bandwidth metric to the Cloudsmith Datadog integration, allowing users to monitor minute-level bandwidth usage via the Cloudsmith v2 analytics endpoint. The metric is disabled by default and can be enabled through configuration. Documentation, configuration, code, and tests have all been updated to support and verify this new feature.

**Realtime Bandwidth Metric Feature:**

* Added support for the `cloudsmith.bandwidth_bytes_interval` metric, which tracks bytes downloaded over the most recent analytics interval using Cloudsmith v2 analytics endpoints. The metric is disabled by default and can be enabled with the `enable_realtime_bandwidth: true` configuration option. Internal defaults (1-minute interval, 120-minute look-back, 300s refresh, minimum 2 points) are used for metric collection. [[1]](diffhunk://#diff-0437399481ad267ab94403e0e25223994bb6d56dfd4a79cb213f271bc3137b61R89-R98) [[2]](diffhunk://#diff-0437399481ad267ab94403e0e25223994bb6d56dfd4a79cb213f271bc3137b61L111-R177) [[3]](diffhunk://#diff-0437399481ad267ab94403e0e25223994bb6d56dfd4a79cb213f271bc3137b61R632-R639) [[4]](diffhunk://#diff-0437399481ad267ab94403e0e25223994bb6d56dfd4a79cb213f271bc3137b61R755-R761) [[5]](diffhunk://#diff-e7216034ba087cc9a0db45fc490da278af8d1fa0aada7ea8c91c5e27f966f317R29) [[6]](diffhunk://#diff-95b2436bf8fbc294c452b0375b26e4e91b105e5a2f77760ef65bccabc86c4ef6R31-R36)

**Documentation Updates:**

* Updated `README.md` and `CHANGELOG.md` to document the new metric, its configuration, and usage instructions, including example YAML configuration and installation version bump. [[1]](diffhunk://#diff-42d77a6be34a4fbf108724d62c4d826c6db292d589f4ae18a73d0cd14181a1d4L28-R33) [[2]](diffhunk://#diff-61997a17f78191e74a351a1b7fef966b98c245c6f4e51fedc72aaddebb3b0884R14-R15) [[3]](diffhunk://#diff-61997a17f78191e74a351a1b7fef966b98c245c6f4e51fedc72aaddebb3b0884L28-R30) [[4]](diffhunk://#diff-61997a17f78191e74a351a1b7fef966b98c245c6f4e51fedc72aaddebb3b0884R39-R48)

**Versioning:**

* Bumped package version to `1.2.0` to reflect the new feature addition.

**Testing:**

* Added tests to verify the correct emission of the real-time bandwidth metric, including edge cases where insufficient data points are present.

**API Changes:**

* Introduced new API endpoint constant `ANALYTICS_METRICS_CLIENT_TIME_SERIES` for accessing Cloudsmith v2 analytics time-series data.### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [ ] If this PR includes a log pipeline, please add a description describing the remappers and processors. 

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: cloudsmith/CHANGELOG.md [hunk 1]
```diff
@@ -25,4 +25,9 @@
 
 ***Added***:
 
-* Initial Release
+## 0.0.3 / 2025-10-29
```
Comment (dd-dominic) [inline cloudsmith/CHANGELOG.md]: Add this to the top as `1.2.0`. Entries should be in descending order.

File: cloudsmith/assets/dashboards/cloudsmith_overview.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "title": "Cloudsmith Overview",
-  "description": "Monitor your Cloudsmith repository usage, member activity, and security posture with key metrics like storage/bandwidth consumption, token usage, and policy violations. Ideal for tracking plan limits and enforcing best practices across your organization.",
+  "description": "Monitor your Cloudsmith repository usage, member activity, and security posture with key metrics like storage/bandwidth consumption, token usage, and policy violations. Ideal for tracking plan limits and enforcing best practices across your organization. (cloned)",
```
Comment (dd-dominic) [inline cloudsmith/assets/dashboards/cloudsmith_overview.json]:
```suggestion
  "description": "Monitor your Cloudsmith repository usage, member activity, and security posture with key metrics like storage/bandwidth consumption, token usage, and policy violations. Ideal for tracking plan limits and enforcing best practices across your organization.",
```

File: cloudsmith/assets/dashboards/cloudsmith_overview.json [hunk 3]
```diff
@@ -423,42 +360,82 @@
...
+            "layout": { "x": 8, "y": 4, "width": 4, "height": 2 }
+          }
+        ]
+      },
+      "layout": { "x": 0, "y": 5, "width": 12, "height": 7 }
+    },
+    {
+      "id": 3721437193949574,
+      "definition": {
+        "title": "Realtime Bandwidth (24h)",
```
Comment (dd-dominic) [inline cloudsmith/assets/dashboards/cloudsmith_overview.json:372]: Edit group to use emphasis style font/header

File: cloudsmith/assets/dashboards/cloudsmith_overview.json [hunk 4]
```diff
@@ -472,82 +449,93 @@
...
+            "layout": { "x": 3, "y": 0, "width": 9, "height": 2 }
+          }
+        ]
+      },
+      "layout": { "x": 0, "y": 12, "width": 12, "height": 3 }
+    },
+    {
+      "id": 4921355289879395,
+      "definition": {
+        "title": "Quota 7d Bars",
```
Comment (dd-dominic) [inline cloudsmith/assets/dashboards/cloudsmith_overview.json:465]: Edit group to use emphasis style font/header

File: cloudsmith/CHANGELOG.md [hunk 5]
```diff
@@ -25,4 +25,9 @@
...
+
+### Added
+
+* Metric: `cloudsmith.bandwidth_bytes_interval` for real-time bandwidth tracking, through Cloudsmith v2 analytics endpoint (disabled by default; enable with `enable_realtime_bandwidth: true`). Uses fixed internal defaults (1-minute interval, aggregate `BYTES_DOWNLOADED_SUM`, look-back 120m, refresh 300s, min points 2).
```
Comment (dd-dominic) [inline cloudsmith/CHANGELOG.md]:
```suggestion
* Added real-time bandwidth metric (`cloudsmith.bandwidth_bytes_interval`) and related dashboard widget
```

**Review by dd-dominic — CHANGES_REQUESTED** [review]

File: cloudsmith/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,12 @@
 # CHANGELOG - Cloudsmith
 
+### 1.2.0 / 2025-11-03
+
+### Added
+
+* Added real-time bandwidth metric (`cloudsmith.bandwidth_bytes_interval`) and related dashboard widget
+* Dashboard widget for realtime interval bytes.
```
Comment (dd-dominic) [inline cloudsmith/CHANGELOG.md]: Redundant as it's already included in the first bullet
```suggestion
```

**Comment by dd-dominic** [comment]
Ignore the ci validation - we can merge without it passing for now

**Comment by dd-dominic** [comment]
Waiting on agent team to take a look

File: cloudsmith/datadog_checks/cloudsmith/check.py [hunk 7]
```diff
@@ -108,8 +119,62 @@ def get_full_path(self, path):
...
+    def get_realtime_bandwidth_info(self):
+        url = self.build_analytics_url()
+        try:
+            return self.get_api_json(url)
+        except Exception:
+            return None
+
+    def parse_realtime_bandwidth(self, response_json):
+        result = {"bandwidth_bytes_interval": None}
+        if not response_json:
```
Comment (dkirov-dd) [inline cloudsmith/datadog_checks/cloudsmith/check.py:159]: nit
Is it expected to sometimes get no response?
It seems worthy for a debug log at least.

File: cloudsmith/datadog_checks/cloudsmith/check.py [hunk 8]
```diff
@@ -108,8 +119,62 @@ def get_full_path(self, path):
...
+            start_dt = datetime.utcnow() - timedelta(minutes=self._RT_LOOKBACK_MINUTES)
+        start_str = start_dt.strftime("%Y-%m-%d+%H:%M")
+        aggregate = self._RT_AGGREGATE
+        return self.build_analytics_base() + f"?interval={interval}&aggregate={aggregate}&start_time={start_str}"
+
+    def get_realtime_bandwidth_info(self):
+        url = self.build_analytics_url()
+        try:
+            return self.get_api_json(url)
+        except Exception:
```
Comment (dkirov-dd) [inline cloudsmith/datadog_checks/cloudsmith/check.py]: request
The call fails silently if an exception is raised. This should not be the case because failure here means missing metrics.
Please add logging.

File: cloudsmith/datadog_checks/cloudsmith/check.py [hunk 9]
```diff
@@ -108,8 +119,62 @@ def get_full_path(self, path):
...
+            return None
+
+    def parse_realtime_bandwidth(self, response_json):
+        result = {"bandwidth_bytes_interval": None}
+        if not response_json:
+            return result
+        results = response_json.get("results") or []
+        if not results:
```
Comment (dkirov-dd) [inline cloudsmith/datadog_checks/cloudsmith/check.py:163]: nit
Same question here, is this normal behavior?
Is it worth adding a log if there are no results?

File: cloudsmith/README.md [hunk 10]
```diff
@@ -34,6 +36,16 @@ For Agent v7.21+ / v6.21+, follow the instructions below to install the Cloudsmi
...
 1. Edit the `cloudsmith.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your Cloudsmith performance data. See the [sample cloudsmith.d/conf.yaml][5] for all available configuration options.
 
+    Example snippet enabling realtime bandwidth:
+
+    ```yaml
+     - url: https://api.cloudsmith.io/v1
+       cloudsmith_api_key: <API-KEY>
+       organization: <ORG-NAME>
+       enable_realtime_bandwidth: true
+       realtime_metric_name: bandwidth.bytes
```
Comment (dkirov-dd) [inline cloudsmith/README.md]:
```suggestion
```
This is mentioned nowhere else.

**Comment by cgarcia768** [comment]
Hi DD Team, we have an Enterprise customer that is urgently requesting this new capability. Are we able to get this update approved?

File: cloudsmith/datadog_checks/cloudsmith/check.py (hunk 7 above)
Comment (BartoszBlizniak) [reply cloudsmith/datadog_checks/cloudsmith/check.py:159]: Hey @dkirov-dd - 

Thanks for taking a look. I'll add logging to this and to the other comment you made. To clarify, we always expect a response; however, if there is no activity, the organisation (which is quite unlikely but possible for a customer who is trailing/occasionally testing during onboarding), the response will come back as an empty array.

File: cloudsmith/README.md (hunk 10 above)
Comment (BartoszBlizniak) [reply cloudsmith/README.md]: Removed

File: cloudsmith/datadog_checks/cloudsmith/check.py (hunk 9 above)
Comment (BartoszBlizniak) [reply cloudsmith/datadog_checks/cloudsmith/check.py:163]: Logging added (responded to question in first comment)

File: cloudsmith/datadog_checks/cloudsmith/check.py (hunk 8 above)
Comment (BartoszBlizniak) [reply cloudsmith/datadog_checks/cloudsmith/check.py]: Logging added
